app/utilities/protocol_importers/protocols_io/v3/protocol_normalizer.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module ProtocolImporters
+  module ProtocolsIO
+    module V3
+      class ProtocolNormalizer < ProtocolImporters::ProtocolNormalizer
+        def load_protocol(id)
+          response = ProtocolImporters::ProtocolsIO::V3::ApiClient.new.single_protocol(id).parsed_response
+
+          normalized_hash = normalize response[:protocol]
+          normalized_hash
+        end
+
+        private
+
+        def normalize(protocol_hash)
+          normalized_data = {}
+          normalized_data[:source_id] = protocol_hash[:id]
+          normalized_data[:name] = protocol_hash[:title]
+          normalized_data[:description] = santz(protocol_hash[:description])
+          normalized_data[:authors] = protocol_hash[:authors].map { |e| e[:name] }.join(', ')
+
+          normalized_data[:steps] = protocol_hash[:steps].map do |e|
+            {
+              source_id: e[:id],
+              name: e[:components].find { |o| o[:type_id] == 6 }[:source][:title],
+              description: santz(e[:components].find { |o| o[:type_id] == 1 }[:source][:description]),
+              position: e[:previous_id].nil? ? 0 : nil
+            }
+          end
+
+          # set positions
+          first_step_id = normalized_data[:steps].find { |s| s[:position].zero? }[:source_id]
+          next_step_id = protocol_hash[:steps].find { |s| s[:previous_id] == first_step_id }.try(:[], :id)
+          current_position = 0
+          while next_step_id
+            current_position += 1
+            normalized_data[:steps].find { |s| s[:source_id] == next_step_id }[:position] = current_position
```

Comment:
This is unnecessary O(n^2) time complexity for searching. Create hash above which maps `previous_id: step_hash` and then have O(1) for searching element.